code/parachain/frame/cosmwasm/cli/src/substrate/cosmwasm.rs

```diff
@@ -0,0 +1,171 @@
+	#[arg(short, long, value_parser = parse_funds)]
+	pub funds: Option<BTreeMap<u128, u128>>,
+	/// Gas limit
+	#[arg(short, long)]
+	pub gas: u64,
+	/// Instantiate message
+	#[arg(short, long)]
+	pub message: String,
+}
+
+#[derive(Args, Debug)]
+pub struct Execute {
+	/// Contract to be executed
+	#[arg(short, long)]
+	pub contract: AccountId32,
+	/// Funds to be moved prior to execution. Format is "ASSET-1:AMOUNT-1,ASSET-2:AMOUNT-2"
+	#[arg(short, long, value_parser = parse_funds)]
+	pub funds: Option<BTreeMap<u128, u128>>,
+	/// Gas limit
+	#[arg(short, long)]
+	pub gas: u64,
+	/// Execute message
+	#[arg(short, long)]
+	pub message: String,
+}
+
+#[derive(Args, Debug)]
+pub struct Migrate {
+	/// Contract to be migrated
+	#[arg(short, long)]
+	pub contract: AccountId32,
+	/// The new code ID to use
+	#[arg(short, long)]
+	pub new_code_id: u64,
+	/// Gas limit
+	#[arg(short, long)]
+	pub gas: u64,
+	/// Migrate message
+	#[arg(short, long)]
+	pub message: String,
+}
+
+#[derive(Args, Debug)]
+#[clap(group(
+    ArgGroup::new("admin")
+    .required(true)
+    .args(&["new_admin", "no_admin"]),
+))]
+pub struct UpdateAdmin {
+	/// Contract to be updated
+	#[arg(short, long)]
+	pub contract: AccountId32,
+	/// New admin of the contract
+	#[arg(short = 'a', long, conflicts_with = "no_admin")]
+	pub new_admin: Option<AccountId32>,
+	/// Erase the admin of the contract (migrates are not possible after this point)
+	// NOTE: This argument won't be used programmatically. It exists so that
+	// users don't accidentally delete the admin because they forget to set
+	// `new_admin`. If this flag is on, we ensure `new_admin` is `None` and
+	// hence the admin will be deleted.
+	#[arg(long, conflicts_with = "new_admin")]
+	pub no_admin: bool,
+	/// Gas limit
+	#[arg(short, long)]
+	pub gas: u64,
+}
+
+#[derive(Args, Debug)]
+pub struct Query {
+	/// Contract to be queried
+	#[arg(short, long)]
+	pub contract: AccountId32,
+	/// Gas limit
+	#[arg(short, long)]
+	pub gas: u64,
+	/// Query request
+	#[arg(short, long)]
+	pub query: String,
+}
+
+impl Upload {
+	pub async fn fetch_code(&self) -> anyhow::Result<Vec<u8>> {
+		let code = if let Some(file_path) = self.file_path.as_ref() {
+			let mut f = fs::File::open(file_path)?;
+			let metadata = fs::metadata(file_path)?;
+			let mut buffer = vec![0u8; metadata.len() as usize];
+			f.read_exact(&mut buffer)?;
+			buffer
+		} else if let Some(url) = self.url.as_ref() {
+			FileFetcher::from_url(url).await.map_err(|e| anyhow!("{}", e))?
+		} else if let Some(cosmos_url) = self.cosmos_rpc.as_ref() {
+			if let Some(contract) = self.contract.as_ref() {
+				CosmosFetcher::from_contract_addr(cosmos_url, contract)
+					.await
+					.map_err(|e| anyhow!("{}", e))?
+			} else if let Some(code_id) = self.code_id.as_ref() {
+				CosmosFetcher::from_code_id(cosmos_url, *code_id)
+					.await
+					.map_err(|e| anyhow!("{}", e))?
+			} else {
```

Comment:
can you replace this else with `expect` then?

Reply:
It was 100% tool as is. Let recallect these notes, and consider on next iterations of changes.

BTW I see CosmosFetcher as garbage approach. It will not work with HTTP proxies/VPNs, etc. Nobody does that. Imho we may end up half code is garbage. Specifically code of such kind no coded for wasm feels will be thown away. Let wait what contract writers will tell.